contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -420,27 +420,27 @@ library DEXAdapter {
         require(_path.length == 2, "ExchangeIssuance: CURVE_WRONG_PATH_LENGTH");
         (int128 i, int128 j) = _getCoinIndices(_pool, _path[0], _path[1], ICurveAddressProvider(_addresses.curveAddressProvider));
 
-        uint256 amountIn = _getAmountInCurve(
-            _pool,
-            i,
-            j,
-            _amountOut,
-            _addresses
-        );
-        require(amountIn <= _maxAmountIn, "ExchangeIssuance: CURVE_OVERSPENT");
 
         if(_path[0] == ETH_ADDRESS){
-            IWETH(_addresses.weth).withdraw(amountIn);
+            IWETH(_addresses.weth).withdraw(_maxAmountIn);
         }
 
-        uint256 returnedAmountOut = _exchangeCurve(i, j, _pool, amountIn, _amountOut, _path[0]);
+        uint256 returnedAmountOut = _exchangeCurve(i, j, _pool, _maxAmountIn, _amountOut, _path[0]);
         require(_amountOut <= returnedAmountOut, "ExchangeIssuance: CURVE_UNDERBOUGHT");
 
+        uint256 swappedBackAmountIn;
+        if(returnedAmountOut > _amountOut){
```

Comment:
Should we add a threshold here, so we don't waste gas on a swap of a few wei ? 
If so, should it be a fixed threshold or user provided ? (The latter while being more flexible would break backwards compatibility in the interface)

Reply:
I don't have a problem with the behavior here. My reasoning being that if we go with a fixed threshold, we wouldn't have a way of picking a good number without pretty intricate testing. If we go with allowing the user to decide, we break the interface which slows down the application of this fix. We can optimize for gas later.